prov/efa/src/efa_cq.c

```diff
@@ -160,12 +160,12 @@ ssize_t efa_cq_readfrom(struct fid_cq *cq_fid, void *buf, size_t count,
 		}
 
 		if (src_addr) {
-			av = cq->domain->qp_table[wc.ibv_wc.qp_num &
-				cq->domain->qp_table_sz_m1]->ep->av;
+			qp_num = ibv_wc_read_qp_num(cq->ibv_cq_ex);
+			src_qp = ibv_wc_read_src_qp(cq->ibv_cq_ex);
+			slid = ibv_wc_read_slid(cq->ibv_cq_ex);
+			av = cq->domain->qp_table[qp_num & cq->domain->qp_table_sz_m1]->ep->av;
```

Comment:
so, before the fix, qp_num, src_qp, and slid was actually zero? Which test is failing due to this bug?

Reply:
Yes. all of them were 0.

The `fi_dgram_pingpong` test was failing because of this bug.

Reply:
But `runfabtests.py` did not report the failure due to the bug in `common.py`

Reply:
could you explain why it only impacts dgram?

Reply:
only dgram endpoint call the function `efa_cq_readfrom`.

rdm endpoint call `ofi_cq_read` which call `rxr_ep_progress` then call `ibv_start_poll` and `ibv_end_poll` directly, bypassing `efa_cq_readfrom`.

BTW, we should rename the file `efa_cq.c` to `efa_dgram_cq.c` and all functions in this file `efa_dgram_xxx`.